src/test/CustomCurveHook.sol

```diff
@@ -36,26 +37,17 @@ contract CustomCurveHook is BaseTestHooks {
         PoolKey calldata key,
         IPoolManager.SwapParams calldata params,
         bytes calldata /* hookData **/
-    ) external override onlyPoolManager returns (bytes4, int128) {
+    ) external override onlyPoolManager returns (bytes4, HookReturnDelta) {
         (Currency inputCurrency, Currency outputCurrency, uint256 amount) = _getInputOutputAndAmount(key, params);
 
         // this "custom curve" is a line, 1-1
         // take the full input amount, and give the full output amount
         manager.take(inputCurrency, address(this), amount);
         outputCurrency.settle(manager, address(this), amount, false);
 
-        // return -amountSpecified to no-op the concentrated liquidity swap
-        return (IHooks.beforeSwap.selector, int128(-params.amountSpecified));
-    }
-
-    function afterSwap(
-        address, /* sender **/
-        PoolKey calldata, /* key **/
-        IPoolManager.SwapParams calldata params,
-        BalanceDelta, /* delta **/
-        bytes calldata /* hookData **/
-    ) external view override onlyPoolManager returns (bytes4, int128) {
-        return (IHooks.afterSwap.selector, int128(params.amountSpecified));
+        // return -amountSpecified as specified to no-op the concentrated liquidity swap
+        HookReturnDelta hookDelta = toHookReturnDelta(int128(-params.amountSpecified), int128(params.amountSpecified));
+        return (IHooks.beforeSwap.selector, hookDelta);
```

Comment:
curious of the gas savings for custom curves 🤔 

but overall i like how we dont need afterSwap now!